### PR TITLE
Youngestdev/compose env

### LINKED ISSUE #1890: Support `$COMPOSE_FILE` in `stack deploy`
**Is your feature request related to a problem? Please describe.**

We have a bunch of microservices with a separate compose YAML file per service, which combine to form a single docker-compose stack. Docker-compose supports an environment variable `$COMPOSE_FILE` that is a path-separated list of compose manifests to be combined to form the stack. This works just like `docker-compose -f` but greatly simplifies CLI interaction when you have a large number of files (we have 30+).

https://docs.docker.com/compose/reference/envvars/#compose_file

**Describe the solution you'd like**

Add support for $COMPOSE_FILE that works like `okteto stack deploy -f`.

**Describe alternatives you've considered**

As mentioned, `-f` works as a workaround but is very unwieldy when there's a large number of files.

### PR DESCRIPTION
**Pull request by Youngestdev**
Fixes #1890

## Proposed changes

- Added support for `$COMPOSE_FILE` env variable.
- Deploy multiple stack files listed in the `COMPOSE_FILE` env variable.

### CODE REVIEW

File: cmd/stack/deploy.go [hunk 1]
```diff
@@ -26,6 +29,13 @@ import (
 	"github.com/spf13/cobra"
 )
 
+func splitComposeFileEnv(value string) []string {
```
Comment (pchico83) [inline cmd/stack/deploy.go:32]: could you add a unit test for this function?

File: cmd/stack/deploy.go [hunk 2]
```diff
@@ -42,6 +52,12 @@ func Deploy(ctx context.Context) *cobra.Command {
 				}
 			}
 
+			composeEnv, present := os.LookupEnv(model.ComposeFile)
```
Comment (pchico83) [inline cmd/stack/deploy.go:55]: @jLopezbarb what would be a good way to test this?

File: pkg/model/const.go [hunk 3]
```diff
@@ -282,4 +282,6 @@ const (
 
 	// GithubRepositoryEnvVar defines the server to be used
 	GithubServerURLEnvVar = "GITHUB_SERVER_URL"
+
+	ComposeFile = "COMPOSE_FILE"
```
Comment (jLopezbarb) [inline pkg/model/const.go:286]:
```suggestion
	ComposeFileEnvVar = "COMPOSE_FILE"
```

File: cmd/stack/deploy.go (hunk 2 above)
Comment (jLopezbarb) [reply cmd/stack/deploy.go:55]: We can encapsulate this code into a function that receives stackPath as parameter and check what happen if:
1. Stackpath is empty and compose file is set
2. Stackpath is empty and compose file is set with more than one path
3. Stackpath is set and compose file is set

Comment (Youngestdev) [reply cmd/stack/deploy.go:55]: Doesn't this block of code doesn't cover the points 1 and 2 🤔 ?

```go
if len(options.StackPath) == 0 && present {
				options.StackPath = splitComposeFileEnv(composeEnv)
			}

Comment (Youngestdev) [reply cmd/stack/deploy.go:55]: I'll rewrite as a function and update PR.

**Comment by pchico83** [comment]
@Youngestdev what's the state of this PR?

**Comment by Youngestdev** [comment]
I'm going to look into writing the tests for the PR. 

Javier & I were supposed to look into it but he's on PTO atm. @pchico83

**Comment by pchico83** [comment]
@jLopezbarb there are pending conflicts

**Review by pchico83 — APPROVED** [review]

**Pull request merged**